convert.c

```diff
@@ -322,6 +322,13 @@ static void trace_encoding(const char *context, const char *path,
 {
 	static struct trace_key coe = TRACE_KEY_INIT(WORKING_TREE_ENCODING);
 	struct strbuf trace = STRBUF_INIT;
+
+	// If tracing is not on, exit early to avoid high memory footprint
+	if (!trace_pass_fl(&coe))
+	{
+		return;
+	}
+
 	int i;
```

Comment:
Please move the `if` block after the `int i` declaration. Also, to avoid being criticized on the Git mailing list, please move the opening `{` curly onto the same line as the `if()`.

Reply:
@HarithaIBM ☝️